File: src/util/md5.rs
//! MD5 工具（兼容 legacy genEncryptedPassword）

use md5::{Digest, Md5};

/// md5 十六进制
pub fn md5_encode(input: &str) -> String {
    let mut hasher = Md5::new();
    hasher.update(input.as_bytes());
    format!("{:x}", hasher.finalize())
}

/// legacy 密码加密：md5(md5(password + salt) + salt)
pub fn gen_encrypted_password(password: &str, salt: &str) -> String {
    md5_encode(&format!("{}{}", md5_encode(&format!("{password}{salt}")), salt))
}

File: internal/util/md5/md5.go
// Package md5 legacy 双 MD5 密码（md5(md5(pwd+salt)+salt)）兼容。
package md5

import (
	"crypto/md5"
	"encoding/hex"
)

// GenEncryptedPassword legacy 双 MD5：md5(md5(pwd+salt)+salt)（小写 hex）。
func GenEncryptedPassword(password, salt string) string {
	h1 := md5.Sum([]byte(password + salt))
	h2 := md5.Sum(append(h1[:], []byte(salt)...))
	return hex.EncodeToString(h2[:])
}
